src/main/proto/netflix/titus/titus_job_api.proto

```diff
@@ -258,6 +265,60 @@ message BasicContainer {
   // (Optional) A collection of system environment variables passed to the
   // container.
   map<string, string> env = 5;
+
+  // (Optional) An array of VolumeMounts. These VolumeMounts will be mounted in
+  // the container, and must reference one of the volumes declared for the Job.
+  // See the k8s docs
+  // https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#volumemount-v1-core
+  // for more technical details.
+  repeated VolumeMount volumeMounts = 6;
+}
+
+// Volumes define some sort of storage for a Task (pod) that is later referenced
+// by individual containers via VolumeMount declarations.
+// https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#volume-v1-core
+// Note that Titus only supports a subset of storage drivers.
+message Volume {
+  // (Required) the name of the volume. This is what is referenced by
+  // VolumeMount requests for individual containers.
+  string name = 1;
+
+  // (Optional) A MainContainerVolume is a volume that exists on the
+  // "main" container, that can be used later via a VolumeMount and shared
+  // with other containers in the task (pod)
+  MainContainerVolume mainContainerVolume = 2;
+}
+
+// VolumeMounts are used to define how to mount a Volume in a container
+// Modeled after k8s volumeMounts:
+// https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#volumemount-v1-core
+message VolumeMount {
+  // (Required) mountPath is the location inside the container where the volume
+  // will be mounted
+  string mountPath = 1;
+
+  // mountPropagation determines how mounts are propagated from the host to
+  // container and the other way around. When not set, MountPropagationNone is
+  // used.
+  string mountPropagation = 2;
+
+  // This must match the Name of a Volume.
+  string volumeName = 3;
+
+  // Mounted read-only if true, read-write otherwise (false or unspecified).
+  // Defaults to false.
+  bool readOnly = 4;
+
+  // Path within the volume from which the container's volume should be mounted.
+  // Defaults to "" (volume's root).
+  string subPath = 5;
+}
+
+message MainContainerVolume {
```

Comment:
This is awesome, but I'm wondering if there's a way we could avoid encoding a "main" container here.  Longer-term, there will be no main container + extra distinction, but we'd still like this as a feature.  Is there a way we could make this an abstraction that will still continue to make sense once it's just a bunch of sibling containers in a pod?

Reply:
Good thinking. 

How about 

```
// Shares a folder from a container to be mounted in other containers
message SharedContainerVolume {
  // Container name of with the folder to export. Usually "main" containers created
  // via the Titus API with the `container` field.
  string sourceContainer = 1;
  string path = 2;
}
```